src/function/memoize.ts

```diff
@@ -6,11 +6,26 @@ export const MEMOIZE_CLEAR_FUNCTION = Symbol('MEMOIZE_CLEAR_FUNCTION');
 /**
  * Memoizes function for passed arguments
  */
-export function memoize<A>(this: any, fn: () => A): () => A;
-export function memoize<A, B>(this: any, fn: (a: A) => B): (a: A) => B;
-export function memoize<A, B, C>(this: any, fn: (a: A, b: B) => C): (a: A, b: B) => C;
-export function memoize<A, B, C, D>(this: any, fn: (a: A, b: B, c: C) => D): (a: A, b: B, c: C) => D;
-export function memoize<A, B, C, D, E>(this: any, fn: (a: A, b: B, c: C, d: D) => E): (a: A, b: B, c: C, d: D) => E {
+export function memoize<A>(this: any, fn: () => A): {
+	(): A
+	[key: string]: () => A
+};
+export function memoize<A, B>(this: any, fn: (a: A) => B): {
+	(a: A): B
+	[key: string]: (a: A) => B
+};
+export function memoize<A, B, C>(this: any, fn: (a: A, b: B) => C): {
+	(a: A, b: B): C
+	[key: string]: (a: A, b: B) => C
+};
+export function memoize<A, B, C, D>(this: any, fn: (a: A, b: B, c: C) => D): {
+	(a: A, b: B, c: C): D
+	[key: string]: (a: A, b: B, c: C) => D
+};
+export function memoize<A, B, C, D, E>(this: any, fn: (a: A, b: B, c: C, d: D) => E): {
+	(a: A, b: B, c: C, d: D): E,
+	[key: string]: E
```

Comment:
Access by index should return a function instead of return type

Reply:
fixed